### DISCUSSION TITLE
Resetting my password takes a long time

### DISCUSSION THREAD

**Discussion by sampl** [npm/feedback#759] [Website] [answered]
I'm sure this is related to the fact that I don't have 2fa (https://github.com/npm/feedback/discussions/753), but is this flow really more secure? I had to verify that I own the email three times.

How I did it:

1. Click sign in
2. Click "forgot password", type my email and submit
3. Click link in email
4. Confirm I forgot my password on the website
5. Copy temporary password
6. Try to sign in with email and temp password, doesn't work
7. Realize I have to sign in with my username and the temp password, that works
8. See prompt to get one-time password from email
9. Go to email, copy OTP
10. Paste OTP in browser
11. Type new password
12. Get prompted to log in with new password
13. See prompt to get one-time password from email
14. Go to email, copy OTP
15. Paste OTP in browser
16. I'm logged in

Preferred behavior:

1. Click sign in
2. Click "forgot password", type my email and submit
3. Click link in email
4. Type new password
6. I'm logged in

Obviously not a major issue, but posting since @MylesBorins said they're open to account usability ideas https://github.com/npm/feedback/discussions/753#discussioncomment-3426619

**Comment by monishcm** [comment, marked as answer]
Thank you for your suggestion. We will investigate this and see how we can improve the experience. I've added this to our backlog.
